public/app/plugins/panel/dashlist/module.tsx

```diff
@@ -0,0 +1,71 @@
+import _ from 'lodash';
+import { PanelModel, PanelPlugin } from '@grafana/data';
+import { DashList } from './DashList';
+import { DashListOptions } from './types';
+import { FolderPicker } from 'app/core/components/Select/FolderPicker';
+import React from 'react';
+import { TagsInput } from '@grafana/ui';
+
+export const plugin = new PanelPlugin<DashListOptions>(DashList)
+  .setPanelOptions(builder => {
+    builder
+      .addBooleanSwitch({
+        path: 'showStarred',
+        name: 'Starred',
+        defaultValue: true,
+      })
+      .addBooleanSwitch({
+        path: 'showRecentlyViewed',
+        name: 'Recently viewed',
+        defaultValue: false,
+      })
+      .addBooleanSwitch({
+        path: 'showSearch',
+        name: 'Search',
+        defaultValue: false,
+      })
+      .addBooleanSwitch({
+        path: 'showHeadings',
+        name: 'Show headings',
+        defaultValue: true,
+      })
+      .addNumberInput({
+        path: 'maxItems',
+        name: 'Max items',
+        defaultValue: 10,
+      })
+      .addTextInput({
+        path: 'query',
+        name: 'Query',
+        defaultValue: '',
+      })
+      .addCustomEditor({
+        path: 'folderId',
+        name: 'Folder',
+        id: 'folder-picker',
+        defaultValue: null,
+        editor: props => {
+          return <FolderPicker initialTitle="All" enableReset={true} onChange={({ id }) => props.onChange(id)} />;
+        },
+      })
+      .addCustomEditor({
+        id: 'tags',
+        path: 'tags',
+        name: 'Tags',
+        description: '',
+        defaultValue: [],
+        editor: props => {
+          return <TagsInput tags={props.value} onChange={props.onChange} />;
+        },
+      });
+  })
+  .setMigrationHandler((panel: PanelModel<DashListOptions> & Record<string, any>) => ({
+    showStarred: panel.options.showStarred ?? panel.starred,
+    showRecentlyViewed: panel.options.showRecentlyViewed ?? panel.recent,
+    showSearch: panel.options.showSearch ?? panel.search,
+    showHeadings: panel.options.showHeadings ?? panel.headings,
+    maxItems: panel.options.maxItems ?? panel.limit,
+    query: panel.options.query ?? panel.query,
+    folderId: panel.options.folderId ?? panel.folderId,
+    tags: panel.options.tags ?? panel.tags,
+  }));
```

Comment:
This needs a little bit more work as it has 2 bugs
* Need to cleanup / remove old properties from panel (using delete keyword).
* Need to only do this if pluginVersion < 7.3 

Both of these will sort of protect against the same bug

We had this bug in the TextPanel the release after it was introduced as then this migration handler is called again, and if the old properties remained it would overwrite the new options settings users had set. 

Se here for reference how it's done for TextPanel (after bug fix)
https://github.com/grafana/grafana/blob/master/public/app/plugins/panel/text/textPanelMigrationHandler.ts#L17